src/logger/index.ts

```diff
@@ -56,28 +56,25 @@ function removeCircular(obj: any, refs: any[] = []): any {
   if (typeof obj !== "object" || !obj) {
     return obj;
   }
-  // If the object defines its own toJSON, prefer that.
-  if (obj.toJSON) {
+  // If the object defines its own toJSON method, use it.
+  if (obj.toJSON && typeof obj.toJSON === "function") {
     return obj.toJSON();
   }
-  if (refs.includes(obj)) {
+  // Only check for circularity among ancestors in the recursion stack.
+  if (refs.indexOf(obj) !== -1) {
     return "[Circular]";
-  } else {
-    refs.push(obj);
   }
-  let returnObj: any;
-  if (Array.isArray(obj)) {
-    returnObj = new Array(obj.length);
-  } else {
-    returnObj = {};
-  }
-  for (const k in obj) {
-    if (refs.includes(obj[k])) {
-      returnObj[k] = "[Circular]";
-    } else {
-      returnObj[k] = removeCircular(obj[k], refs);
+  // Add the current object to the recursion stack.
+  refs.push(obj);
+
+  const returnObj: any = Array.isArray(obj) ? [] : {};
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
```

Comment:
We are no longer doing this logic here:

```
    if (refs.includes(obj[k])) {
      returnObj[k] = "[Circular]";
    } else {
```

Makes the code slightly less efficient?

Maybe more readable, but feels like we're changing how the code works a bit?